### PR TITLE
Feature/bngp 527 cya

### PR DESCRIPTION
**Pull request by reneli5**

### CODE REVIEW

File: packages/webapp/src/routes/land/check-ownership-details.js [hunk 1]
```diff
@@ -1,11 +1,23 @@
...
+    return h.view(constants.views.CHECK_OWNERSHIP_DETAILS, { name, consent, fileName, boolToYesNo, hideConsent })
   },
-  post: async (_request, h) => h.redirect(constants.routes.REGISTER_LAND_TASK_LIST)
+  post: async (request, h) => {
+    processCompletedRegistrationTask(request, { taskTitle: 'Land information', title: 'Add land ownership details' })
+    return h.redirect(constants.routes.REGISTER_LAND_TASK_LIST)
+  }
+}
+
+function getOwneershipFileName (fileLocation) {
```
Comment (pwadmore-ea) [inline packages/webapp/src/routes/land/check-ownership-details.js:19]: Typo - ownership

**Review by pwadmore-ea — CHANGES_REQUESTED** [review]
@reneli5 - I've raised a number of comments but I don't think there is anything major.

- In addition to the comments on individual changes please check the general formatting of packages/webapp/src/views/land/register-land-task-list.html.
- **Observed defect using branch code** - No details shown on check land boundary details page (observed in Chrome and Firefox) 
image
- I suggest making the upload pages neater if they are accessed out of of sequence
image

File: packages/webapp/src/routes/__tests__/land/register-land-task-list.spec.js [hunk 2]
```diff
@@ -31,7 +31,7 @@ describe(url, () => {
       const response = await submitGetRequest(getOptions)
       expect(response.statusCode).toBe(200)
       expect(viewResult).toEqual('land/register-land-task-list')
-      expect(contextResult.registrationTasks.taskList.length).toEqual(4)
+      expect(contextResult.registrationTasks.taskList.length).toEqual(5)
```
[the remark this answers is not in the document]
Comment (reneli5) [reply packages/webapp/src/routes/__tests__/land/register-land-task-list.spec.js:34]: That is managed by the application, it is much of a completion step.

File: packages/webapp/src/routes/__tests__/land/check-legal-agreement-details.spec.js [hunk 3]
```diff
@@ -108,104 +109,104 @@ describe(url, () => {
...
-          })
-          const request = {
-            yar: redisMap
-          }
-          const h = {
-            redirect: (view) => {
-              viewResult = view
-            }
+  })
+  it('should continue with the flow with other party role name', done => {
```
[the remark this answers is not in the document]
Comment (reneli5) [reply packages/webapp/src/routes/__tests__/land/check-legal-agreement-details.spec.js]: Refactored

File: packages/webapp/src/routes/land/check-geospatial-land-boundary-file.js [hunk 4]
```diff
@@ -16,7 +16,7 @@ const handlers = {
     let route
     switch (request.payload.confirmGeospatialLandBoundary) {
       case constants.confirmLandBoundaryOptions.YES:
-        route = constants.routes.CONFIRM_GEOSPATIAL_LAND_BOUNDARY
+        route = constants.routes.CHECK_LAND_BOUNDARY_DETAILS
```
[the remark this answers is not in the document]
Comment (reneli5) [reply packages/webapp/src/routes/land/check-geospatial-land-boundary-file.js:19]: This flow need to be covered by this ticket: https://eaflood.atlassian.net/browse/BNGP-526

File: packages/webapp/src/routes/land/landowner-consent.js [hunk 5]
```diff
@@ -8,7 +8,7 @@ const handlers = {
   post: async (request, h) => {
     const consent = request.payload.landownerConsent
     const name = request.yar.get(constants.redisKeys.FULL_NAME)
-    if (!consent || !JSON.parse(consent)) {
+    if (!consent) {
```
[the remark this answers is not in the document]
Comment (reneli5) [reply packages/webapp/src/routes/land/landowner-consent.js:11]: landownerConsent will either be true or undefined. Don't see any need for parsing it.

File: packages/webapp/src/routes/land/legal-agreement-type.js [hunk 6]
```diff
@@ -26,7 +25,7 @@ const handlers = {
   },
   post: async (request, h) => {
     const legalAgreementType = request.payload.legalAgreementType
-    if (legalAgreementType) {
+    if (legalAgreementType !== undefined) {
```
[the remark this answers is not in the document]
Comment (reneli5) [reply packages/webapp/src/routes/land/legal-agreement-type.js:28]: Not needed. I have reverted it.

**Review by pwadmore-ea — APPROVED** [review]
Unable to reproduce blank land boundary check your details locally again. If I can reproduce it at some point it can be investigated then.
